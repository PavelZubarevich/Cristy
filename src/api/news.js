const news = [
  {
    id: 1,
    title: 'Get 20% OFF on Nike Air Uomo',
    text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut',
    date: '2022-10-30T16:31:00Z',
    image: 'https://as2.ftcdn.net/v2/jpg/04/19/14/17/1000_F_419141782_2mv5HVpvCW4IbixYdJOasaUZfQVfaMAS.jpg',
  },
  {
    id: 2,
    title: 'Get 20% OFF on Nike Air Uomo',
    text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut',
    date: '2022-09-30T16:31:00Z',
    image: 'https://as2.ftcdn.net/v2/jpg/04/19/14/17/1000_F_419141782_2mv5HVpvCW4IbixYdJOasaUZfQVfaMAS.jpg',
  },
  {
    id: 3,
    title: 'Get 20% OFF on Nike Air Uomo',
    text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut',
    date: '2022-08-30T16:31:00Z',
    image: 'https://as2.ftcdn.net/v2/jpg/04/19/14/17/1000_F_419141782_2mv5HVpvCW4IbixYdJOasaUZfQVfaMAS.jpg',
  },
];

// eslint-disable-next-line
export const getLastNews = () => news;
