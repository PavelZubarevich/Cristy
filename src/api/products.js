import axios from 'axios';

export const getAllProducts = async () => {
  try {
    const response = await axios.get('https://dummyjson.com/products');
    return response.data;
  } catch (e) {
    console.error(e);
    return [];
  }
};

export const getProducts = async (limit) => {
  try {
    const response = await axios.get(`https://dummyjson.com/products?limit=${limit}`);
    return response.data;
  } catch (e) {
    console.error(e);
    return [];
  }
};
