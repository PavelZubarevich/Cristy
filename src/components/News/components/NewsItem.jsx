import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

function NewsItem({
  id, title, image, description, date,
}) {
  const { t } = useTranslation();

  const newsDate = new Date(date);
  const shortMonthName = new Intl.DateTimeFormat('en-US', { month: 'short' }).format;

  return (
    <div className="newsItem">
      <div className="newsItem__image">
        <img src={image} alt={title} />
      </div>
      <div className="newsItem__content">
        <div>
          <div className="newsItem__title">{title}</div>
          <div className="newsItem__description">{description}</div>
        </div>
        <Link to={`/news${id}`} className="newsItem__link">
          {t('ReadMoreLabel')}
          ...
        </Link>
      </div>
      <div className="newsItem__date">
        <div>{newsDate.getDate()}</div>
        <div>{shortMonthName(newsDate)}</div>
      </div>
    </div>
  );
}

export default NewsItem;

NewsItem.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  date: PropTypes.string,
};

NewsItem.defaultProps = {
  title: '',
  description: '',
  image: '',
  date: '',
};
