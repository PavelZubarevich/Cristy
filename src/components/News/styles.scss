@import '../../styles/vars.scss';

.news {
  &__content {
    padding: 90px 0;
    display: flex;
    margin: 0 -15px;
  }

  &__item {
    flex: 1 1 33%;
    padding: 0 15px;
  }
}

.newsItem {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;

  &__image {
    height: 230px;
    margin-bottom: 20px;

    &>img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }

  &__content {
    flex: 1;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

  &__title {
    font-family: "Lora";
    font-weight: 700;
    font-size: 18px;
    color: $secondaryDarkDark;
    margin-bottom: 10px;
  }

  &__description {
    font-family: "Raleway";
    font-weight: 400;
    font-size: 14px;
    color: $primaryText;
    margin-bottom: 10px;
  }

  &__link {
    font-family: "Lora";
    font-weight: 700;
    font-size: 16px;
    color: $primary;
  }

  &__date {
    position: absolute;
    top: 20px;
    left: 20px;
    background-color: $primary;
    width: 50px;
    height: 50px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    &>div {
      font-family: "Lora";
      font-weight: 700;
      font-size: 14px;
      color: #fff;
    }
  }
}