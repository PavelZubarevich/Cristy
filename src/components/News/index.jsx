import { getLastNews } from '../../api/news';
import NewsItem from './components/NewsItem';
import './styles.scss';

function News() {
  const news = getLastNews();

  return (
    <div className="news">
      <div className="container">
        <div className="news__content">
          {news.map((elem) => (
            <div className="news__item" key={elem.id}>
              <NewsItem
                id={elem.id}
                title={elem.title}
                description={elem.text}
                image={elem.image}
                date={elem.date}
              />
            </div>
          ))}
        </div>
      </div>

    </div>
  );
}

export default News;
