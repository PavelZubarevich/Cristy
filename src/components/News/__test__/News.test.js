import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import News from '..';
import NewsItem from '../components/NewsItem';
import { getLastNews } from '../../../api/news';

const news = [
  {
    id: 1,
    title: 'Get 20% OFF on Nike Air Uomo',
    text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut',
    date: '2022-10-30T16:31:00Z',
    image: 'https://as2.ftcdn.net/v2/jpg/04/19/14/17/1000_F_419141782_2mv5HVpvCW4IbixYdJOasaUZfQVfaMAS.jpg',
  },
  {
    id: 2,
    title: 'Get 20% OFF on Nike Air Uomo',
    text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut',
    date: '2022-09-30T16:31:00Z',
    image: 'https://as2.ftcdn.net/v2/jpg/04/19/14/17/1000_F_419141782_2mv5HVpvCW4IbixYdJOasaUZfQVfaMAS.jpg',
  },
  {
    id: 3,
    title: 'Get 20% OFF on Nike Air Uomo',
    text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut',
    date: '2022-08-30T16:31:00Z',
    image: 'https://as2.ftcdn.net/v2/jpg/04/19/14/17/1000_F_419141782_2mv5HVpvCW4IbixYdJOasaUZfQVfaMAS.jpg',
  },
];

jest.mock('../../../api/news', () => ({
  getLastNews: jest.fn(),
}));

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str) => str,
    i18n: {
      changeLanguage: () => new Promise(() => { }),
      options: {
        supportedLngs: ['ru', 'en', 'cimode'],
      },
    },
  }),
}));

describe('News', () => {
  it('should render with news', () => {
    getLastNews.mockImplementation(() => news);
    render(
      <News />,
      { wrapper: BrowserRouter },
    );
    expect(document.body).toMatchSnapshot();
  });

  it('should render without news', () => {
    getLastNews.mockImplementation(() => []);
    render(
      <News />,
      { wrapper: BrowserRouter },
    );
    expect(document.body).toMatchSnapshot();
  });
});

describe('News item', () => {
  it('should render news item', () => {
    render(
      <NewsItem
        id={1}
        title="title"
        description="description"
        image=""
        date="2022-09-30T16:31:00Z"
      />,
      { wrapper: BrowserRouter },
    );
    expect(document.body).toMatchSnapshot();
  });
});
