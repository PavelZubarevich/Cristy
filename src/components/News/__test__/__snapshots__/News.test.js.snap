// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News item should render news item 1`] = `
<body>
  <div>
    <div
      class="newsItem"
    >
      <div
        class="newsItem__image"
      >
        <img
          alt="title"
          src=""
        />
      </div>
      <div
        class="newsItem__content"
      >
        <div>
          <div
            class="newsItem__title"
          >
            title
          </div>
          <div
            class="newsItem__description"
          >
            description
          </div>
        </div>
        <a
          class="newsItem__link"
          href="/news1"
        >
          ReadMoreLabel
          ...
        </a>
      </div>
      <div
        class="newsItem__date"
      >
        <div>
          30
        </div>
        <div>
          Sep
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`News should render with news 1`] = `
<body>
  <div>
    <div
      class="news"
    >
      <div
        class="container"
      >
        <div
          class="news__content"
        >
          <div
            class="news__item"
          >
            <div
              class="newsItem"
            >
              <div
                class="newsItem__image"
              >
                <img
                  alt="Get 20% OFF on Nike Air Uomo"
                  src="https://as2.ftcdn.net/v2/jpg/04/19/14/17/1000_F_419141782_2mv5HVpvCW4IbixYdJOasaUZfQVfaMAS.jpg"
                />
              </div>
              <div
                class="newsItem__content"
              >
                <div>
                  <div
                    class="newsItem__title"
                  >
                    Get 20% OFF on Nike Air Uomo
                  </div>
                  <div
                    class="newsItem__description"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut
                  </div>
                </div>
                <a
                  class="newsItem__link"
                  href="/news1"
                >
                  ReadMoreLabel
                  ...
                </a>
              </div>
              <div
                class="newsItem__date"
              >
                <div>
                  30
                </div>
                <div>
                  Oct
                </div>
              </div>
            </div>
          </div>
          <div
            class="news__item"
          >
            <div
              class="newsItem"
            >
              <div
                class="newsItem__image"
              >
                <img
                  alt="Get 20% OFF on Nike Air Uomo"
                  src="https://as2.ftcdn.net/v2/jpg/04/19/14/17/1000_F_419141782_2mv5HVpvCW4IbixYdJOasaUZfQVfaMAS.jpg"
                />
              </div>
              <div
                class="newsItem__content"
              >
                <div>
                  <div
                    class="newsItem__title"
                  >
                    Get 20% OFF on Nike Air Uomo
                  </div>
                  <div
                    class="newsItem__description"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut
                  </div>
                </div>
                <a
                  class="newsItem__link"
                  href="/news2"
                >
                  ReadMoreLabel
                  ...
                </a>
              </div>
              <div
                class="newsItem__date"
              >
                <div>
                  30
                </div>
                <div>
                  Sep
                </div>
              </div>
            </div>
          </div>
          <div
            class="news__item"
          >
            <div
              class="newsItem"
            >
              <div
                class="newsItem__image"
              >
                <img
                  alt="Get 20% OFF on Nike Air Uomo"
                  src="https://as2.ftcdn.net/v2/jpg/04/19/14/17/1000_F_419141782_2mv5HVpvCW4IbixYdJOasaUZfQVfaMAS.jpg"
                />
              </div>
              <div
                class="newsItem__content"
              >
                <div>
                  <div
                    class="newsItem__title"
                  >
                    Get 20% OFF on Nike Air Uomo
                  </div>
                  <div
                    class="newsItem__description"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut
                  </div>
                </div>
                <a
                  class="newsItem__link"
                  href="/news3"
                >
                  ReadMoreLabel
                  ...
                </a>
              </div>
              <div
                class="newsItem__date"
              >
                <div>
                  30
                </div>
                <div>
                  Aug
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`News should render without news 1`] = `
<body>
  <div>
    <div
      class="news"
    >
      <div
        class="container"
      >
        <div
          class="news__content"
        />
      </div>
    </div>
  </div>
</body>
`;
